cluster.c

```diff
@@ -95,19 +95,16 @@ char *ShardGroupSerialize(ShardGroup *sg)
 RRStatus ShardGroupDeserialize(const char *buf, size_t buf_len, ShardGroup *sg)
 {
     /* Make a mutable, null terminated copy */
-    char str[buf_len + 1];
-    memcpy(str, buf, buf_len);
-    str[buf_len] = '\0';
-    char *s = str;
-
+    const char *s = buf;
+    char *nl;
+    char *endptr;
     memset(sg, 0, sizeof(*sg));
 
-    /* Find and null terminate shard group id */
-    char *nl = strchr(s, '\n');
-    if (!nl) {
+    /* Find shard group id */
+    nl = strchr(s, '\n');
+    if (!nl || nl - buf > buf_len) {
```

Comment:
Just fyi, you can use `memchr()` if you're worried buf may not have '\0' at the end. Because if there is no '\0' at the end, if `nl - buf > buf_len`  happens, means `strchr()` went beyond `buf_len`, it may trigger segfault.

Reply:
ok, will switch it to that.